### PR TITLE
Monitor server status

### PR DESCRIPTION
**Pull request by kate-v-stepanova**

### CODE REVIEW

**Comment by vezzi** [comment]
Looks fine to me, 
not clear how you pass server names and gdocs credentials with the configurations files, can you copy and paste the new config file (without passwords)?

**Comment by kate-v-stepanova** [comment]
1. **credentials file** must be generated on https://console.developers.google.com
In our case it's already generated, but according to google policy, we are allowed to use the only copy of it, so we can download it only once, at the moment when it's generated as it contains private key.

So, when we set up the script on our server, **we have to generate credentials file again**. To do that, go to:  https://console.developers.google.com/project/server-status-1047/apiui/credential, click on the email address (the only one in the list, looks something like blablabla@**developer.gserviceaccount.com** )
then click: **Generate new JSON key** and download it to the taca/server_status/ directory as gdocs_credentials.json 

open console

generate json key

It can be also saved somewhere else, but then make sure to specify the path in config.py file

2. **config file**
I included **config.example.py** which explains a bit how config.py should look like. Now more detailed.
all the fields are mandatory:
USER = "username" -> user which will ssh to the servers, in our case hiseq.bioinfo (hope, it's not a secret information, if it is I can remove it)
SERVERS  = {} - dictionary, where key is url to server, for example: 'b5.something.se', and value is a command that we need to execute to get a number of available space in percent. in our case it is "df -h /home  | awk '{print$(NF-1)}' | tail -n 1"
example:
SERVERS = {
'b5.something.se': "df -h /home  | awk '{print$(NF-1)}' | tail -n 1",
"nas-1234.something.else.se": "df -h /home  | awk '{print$(NF-1)}' | tail -n 1",
"localhost": ""df -h /home  | awk {print$(NF-1)} | tail -n 1" -> the localhost is server from where we run the script. if it's b5, then instead of b5 we write localhost, if it's nas-1234, we write nas-1234
}
G_CREDENTIALS = os.path.join(os.path.dirname(__file__), 'gdocs_credentials.json') -> abs path to gdocs_credentials file that we downloaded in paragraph 1.

G_SHEET = 'HiSeq - NAS connections' -> name of the google sheet (important: file must be shared with email listed here: 
open console

G_SCOPE = ['https://spreadsheets.google.com/feeds'] -> this is static value, we don't change it
G_SHEET_MAP = {	} - dictionary, where key is server name, e.g: b5, and the key is address of cell inside of the document where we insert value, e.g: G22
example:
G_SHEET_MAP = {	'b5': 'D22',
				'nas-10': 'J6',
				}

**Comment by ewels** [comment]
@kate-v-stepanova: I edited your comment a bit with some more markdown, mostly adding some extra backticks ( ` ) to make the code [easier to read](https://help.github.com/articles/markdown-basics/#code-formatting). Hope that's ok!

**Comment by kate-v-stepanova** [comment]
@ewels yes sure, that's great, I'm not too good in making texts well readable. Thanks! 
Will try to do it next time better :)

**Comment by ewels** [comment]
No problem - it's just my OCD. I do it for everyone's comments, I just thought I'd warn you so that you weren't confused why it changed :grinning:

File: taca/server_status/config.example.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import os
```
Comment (vezzi) [inline taca/server_status/config.example.py]: hej @kate-v-stepanova is this file still needed? maybe we shoudl add a better example of configuraition.yaml file in the main TACA folder

File: taca/server_status/server_status.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	result = {}
+	config = CONFIG['server_status']
+	servers = config.get('servers', [])
+	for server_url in servers.keys():
+		# get path of disk
+		path = servers[server_url]
+
+		# get command
+		command = config['command'] + ' ' + path
+
```
Comment (vezzi) [inline taca/server_status/server_status.py]: I think this syntax is preferred
`command = "{} {}".format(config['command'], path)`

right @Galithil ?

File: taca/server_status/server_status.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		output = __parse_output(output)
+
+		try:
+			# remove % symbol and convert string to int
+			used_space = int(output.strip().replace('%', ''))
+			# how many percent available
+			available_space = 100 - used_space
+			result[server_url] = str(available_space) + '%'
+		except:
+			# sometimes it fails for whatever reason as Popen returns not what it is supposed to
```
Comment (vezzi) [inline taca/server_status/server_status.py]: not a big fun of this.... , print to the log a warning and then continue.... however would be better to fail as we the risk is that we update with wrong values. 
Test is on Preproc1, 2, and B5. If it works then it is fine and this line can became an exception to be thrown

File: taca/server_status/server_status.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	output = output.split('\n')[-1] # split by lines and take the last line
+	output = output.strip() # remove spaces
+	output = output.split() # split line by space symbols
+
+	# output = ['24T', '12T', '13T', '49%', '/srv/illumina']
+	for item in output: # select the item containing '%' symbol
+		if '%' in item:
+			return item
+
+	return '' # if no '%' in output, return empty string
```
Comment (vezzi) [inline taca/server_status/server_status.py]: maybe better to rerun NaN, so that it can be noticed from Gdocs

Comment (kate-v-stepanova) [inline taca/server_status/server_status.py]: updated - for all the comments
